module/VuFindSearch/src/VuFindSearch/Backend/Blender/Backend.php

```diff
@@ -0,0 +1,590 @@
+                    if (!isset($activeBackends[$active])) {
+                        throw new \Exception(
+                            "Invalid blender_backend filter: Backend $active not"
+                            . ' enabled'
+                        );
+                    }
+                    $filteredBackends[$active] = $activeBackends[$active];
+                }
+            }
+        }
+        if ($filteredBackends) {
+            $activeBackends = $filteredBackends;
+        }
+        foreach ($fq as $current) {
+            if (strncmp($current, '-blender_backend:', 17) === 0) {
+                $disabled = trim(substr($current, 17), '"');
+                if (isset($activeBackends[$disabled])) {
+                    unset($activeBackends[$disabled]);
+                }
+            }
+        }
+
+        return $activeBackends;
+    }
+
+    /**
+     * Get a record from the given backend by offset
+     *
+     * @param array $backendDetails Details for the backend
+     * @param array $backendRecords Record buffer
+     * @param int   $offset         Record offset
+     * @param int   $blockSize      Blending block size
+     *
+     * @return RecordInterface|null
+     */
+    protected function getRecord(
+        array $backendDetails,
+        array &$backendRecords,
+        int $offset,
+        int $blockSize
+    ): RecordInterface {
+        if (!$backendRecords) {
+            $collection = $backendDetails['backend']->search(
+                $backendDetails['query'],
+                $offset,
+                max($blockSize, 20),
+                $backendDetails['params']
+            );
+            $backendRecords = $collection->getRecords();
+        }
+        return $backendRecords ? array_shift($backendRecords) : null;
+    }
+
+    /**
+     * Get the block size for the given result count
+     *
+     * @param int $resultCount Result count
+     *
+     * @return int
+     */
+    protected function getBlockSize(int $resultCount): int
+    {
+        foreach ($this->adaptiveBlockSizes as $size) {
+            $parts = explode(':', $size, 2);
+            $blockSize = intval($parts[1] ?? 0);
+            if ($blockSize === 0) {
+                throw new \Exception("Invalid adaptive block size: $size");
+            }
+            $rangeParts = explode('-', $parts[0]);
+            $from = intval($rangeParts[0]);
+            $to = intval($rangeParts[1] ?? 0);
+            if ($from > $to) {
+                throw new \Exception("Invalid adaptive block size: $size");
+            }
+            if ($from <= $resultCount && $resultCount <= $to) {
+                return $blockSize;
+            }
+        }
+        return $this->blockSize;
+    }
+
+    /**
+     * Set EventManager instance.
+     *
+     * @param EventManagerInterface $events Event manager
+     *
+     * @return void
+     * @todo   Deprecate `VuFind\Search' event namespace (2.2)
```

Comment:
Any idea what this `@todo` is about? Seems to have been copied and pasted from elsewhere... wondering if it's something we should consider cleaning up rather than replicating.

Reply:
Yes, this looks like it may be a very old mess in need of cleaning up. I've opened [VUFIND-1547](https://vufind.org/jira/browse/VUFIND-1547) to address it.